datasource-decorator-spring-boot-autoconfigure/src/main/java/com/github/gavlyukovskiy/micrometer/DataSourceMetricsHolder.java

```diff
@@ -43,40 +40,37 @@
 
     @Override
     public void bindTo(MeterRegistry registry) {
-        connectionObtainTimer = Timer.builder("data.source.connection.wait")
+        connectionObtainTimer = Timer.builder("data.source.connections.wait")
+                .publishPercentileHistogram()
```

Comment:
Note that this is only useful for Prometheus and Atlas backends. In rc3, we added the ability to configure percentile histograms on specific timers at a registry level and in rc4, we'll add a further variation on this to allow them to be configured via a property. So a user targeting Prometheus could enable histograms via a property with something like this:

```yml
spring.metrics.filter.data.source.connections.*:
   publishPercentileHistogram: true
```

This isn't wrong, just be aware.

Reply:
Awesome! Thanks for the advice.

Reply:
And for the review of course!